src/consumer/component/dependencies/files-dependency-builder/detector-hook.ts

```diff
@@ -28,6 +28,12 @@ export interface DependencyDetector {
    */
   isSupported(context: FileContext): boolean;
 
+  /**
+   * determine what type of content the detector is for.
+   * if no type provided, the type would be the ext.
+   */
+  type?: string;
```

Comment:
What possible values are for this? 
if it's general string, we should improve the comment above it to explain what it is

Reply:
This `type` is what `precinct()` previously receives as the second argument. Also, it's what the previous `getType()` returns in `paperwork()`. I added this field into the detector interface to combine the 2 steps directly.

The previous implementation has 2 steps: 1) get the type from the ext, 2) get detective from the type.

Sometimes ext names and types are not 1:1 matched. For example, `.ts` and `.tsx` both match the type `ts`.

For custom env detectors, I think it's also a good practice for detector authors to specify what type of files it belongs to.


And I updated the code comment a little bit. What do you think now? :-)